drivers/ntc/ntc_ntb_msi.c

```diff
@@ -1806,9 +1799,7 @@ static bool ntc_request_dma(struct ntc_dev *ntc)
 	dma_cap_set(DMA_MEMCPY, mask);
 
 	for (i = 0, j = 0; i < num_dma_chan; i++) {
-		dma = dma_request_channel(mask, ntc_ntb_filter_bus, NULL);
-		ntc->dma_chan[j].chan = dma;
-		if (dma)
+		if (!!(ntc->dma_chan[j].chan = dma_request_chan_by_mask(&mask)))
```

Comment:
We want from one side to use as much channels as we can (to distribute the work and balance the load) and from the other hand to make sure each ring uses the same channel so we will be sure there are no races in the ring (e.g. will transfer the data, update RQP & send vbell all on the same dma channel, but we can use 8 different QPs each on other dma channel so most of the work can be made in parallel), I don't see this code does this.